### PR TITLE
Add document for thread dump API

### PR DESCRIPTION
**Pull request by DengYiping**
for gci2015

### CODE REVIEW

File: html/api.html [hunk 1]
```diff
@@ -1198,19 +1198,142 @@ <h2 class="sub-header">
...
+<ul>
+	<li>Memory usage</li>
+	<li>Total running time</li>
+	<li>Time to restart</li>
+	<li>Status of individual threads</li>
+	<li>Threads that have been terminated</li>
+</ul>
+<p>Thread dump can be used to track the memory use of loklak, and it helps developers to discover potential problems like memory leaks. There are six states that has been assigned to a thread, see further reading: <a href="https://docs.oracle.com/javase/7/docs/api/java/lang/Thread.State.html">Enum Thread.State</a></p>
+<p>At the end of the thread dump is a list of running thread. This list may indicate the load of the server.</p>
+<p>Here is some example of <a href="http://localhost:9000/api/threaddump.txt.">threaddump.txt</a>(The following example has been delibrately cutted to fit in the document):</p>
```
Comment (sudheesh001) [inline html/api.html]: Nit: deliberately `cut`.

File: html/api.html [hunk 2]
```diff
@@ -671,24 +671,24 @@ <h2 class="sub-header">
...
 	To retrieve an image, there are three possible paths: <code>/api/proxy.gif</code>,
     <code>/api/proxy.png</code>, or <code>/api/proxy.jpg</code>.
-        
-	This API supports both GET and POST requests. The HTTP GET or POST request 
+
+	This API supports both GET and POST requests. The HTTP GET or POST request
 	has these attributes:</p>
 <ul class="request">
 	<li><div class="tabcol0">url</div>
 		<div class="tabcol1">&nbsp;=&nbsp;&lt;string&gt</div>//
```
Comment (sudheesh001) [inline html/api.html]: `&gt;` Missed the `;`

File: html/api.html [hunk 3]
```diff
@@ -671,24 +671,24 @@ <h2 class="sub-header">
...
-		the public url of the data source. This URL will be used to load the 
+		the public url of the data source. This URL will be used to load the
 		content.</li>
 	<li><div class="tabcol0">screen_name</div>
-		<div class="tabcol1">&nbsp;=&nbsp;&lt;string&gt</div>// 
+		<div class="tabcol1">&nbsp;=&nbsp;&lt;string&gt</div>//
```
Comment (sudheesh001) [inline html/api.html]: Missed the `;` => `&gt;`

**Comment by DengYiping** [comment]
I fix it. It's not my code though. @sudheesh001 It is highlighted in my text editor.

**Comment by sudheesh001** [comment]
Please squash them into one single commit. Maybe this link could help
https://asciinema.org/a/11269

**Pull request closed without merging**